### PR TITLE
p2p: Disconnect, not discourage, misbehaving NODE_BLOOM peers

### PR DESCRIPTION
**Pull request by maflcko**
This continues the story started in commit 3a10d935ac8ebabdfd336569d943f042ff84b13e

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -3709,20 +3709,16 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
...
         if (vData.size() > MAX_SCRIPT_ELEMENT_SIZE) {
-            bad = true;
+            pfrom.fDisconnect = true;
         } else if (pfrom.m_tx_relay != nullptr) {
             LOCK(pfrom.m_tx_relay->cs_filter);
             if (pfrom.m_tx_relay->pfilter) {
                 pfrom.m_tx_relay->pfilter->insert(vData);
             } else {
-                bad = true;
+                pfrom.fDisconnect = true;
```
Comment (Xekyo) [inline src/net_processing.cpp:3719]: This behavior seems to have previously triggered log messages "too-large bloom filter` or  `bad filteradd message`. I see that in other instances of `pfrom.fDisconect` it seems to be often preceded by log messages, e.g. `LogPrintf("Disconnecting outbound peer %d -- headers chain has insufficient work\n", pfrom.GetId());`.

Should there be a log message here for the disconnect as well?

**Comment by jnewbery** [comment]
Can you add a motivation to the PR description please? :)

**Review by jnewbery — COMMENTED** [review]
I'm not sure if this is an improvement. Setting `fDisconnect` directly bypasses the logic in `MaybeDiscourageAndDisconnect()` that prevents us from disconnecting peers with `PF_NOBAN` or manual connections. Is that intentional?

**Comment by MarcoFalke** [comment]
Sorry, this was opened accidentally (not ready yet)

File: src/net_processing.cpp [hunk 2]
```diff
@@ -3709,20 +3709,16 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
 
         // Nodes must NEVER send a data item > 520 bytes (the max size for a script data object,
         // and thus, the maximum size any matched object can have) in a filteradd message
-        bool bad = false;
         if (vData.size() > MAX_SCRIPT_ELEMENT_SIZE) {
-            bad = true;
+            pfrom.fDisconnect = true;
         } else if (pfrom.m_tx_relay != nullptr) {
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply src/net_processing.cpp:3714]: Actually this is impossible, since we'll never advertise `NODE_BLOOM` for outbound peers.

**Pull request closed without merging**